google/cloud/storage/internal/signed_url_requests.cc

```diff
@@ -80,6 +82,132 @@ std::ostream& operator<<(std::ostream& os, V2SignUrlRequest const& r) {
   return os << "SingUrlRequest={" << r.StringToSign() << "}";
 }
 
+namespace {
+std::string QueryStringFromParameters(
+    CurlHandle& curl,
+    std::multimap<std::string, std::string> const& parameters) {
+  std::string result;
+  char const* sep = "";
+  for (auto const& qp : parameters) {
+    result += sep;
+    result += curl.MakeEscapedString(qp.first).get();
+    result += '=';
+    result += curl.MakeEscapedString(qp.second).get();
+    sep = "&";
+  }
+  return result;
+}
+
+std::string TrimHeaderValue(std::string const& value) {
+  std::string tmp = value;
+  tmp.erase(0, tmp.find_first_not_of(' '));
+  tmp = tmp.substr(0, tmp.find_last_not_of(' ') + 1);
+  auto end = std::unique(tmp.begin(), tmp.end(), [](char a, char b) {
+    if (a != b) {
+      return false;
+    }
+    return a == ' ';
+  });
+  tmp.erase(end, tmp.end());
+  return tmp;
+}
+}  // namespace
+
+void V4SignUrlRequest::AddMissingRequiredHeaders() {
+  auto const& headers = common_request_.extension_headers();
+  if (headers.count("host") == 0) {
```

Comment:
Aside: I'm not a huge fan of using count() == 0 as a proxy for find() != end, but just sayin'.

Reply:
Fixed.